### PR TITLE
TestRequestHandler implementation and relevant cleanup.

### PR DESCRIPTION
**Pull request by harshjain2**
TestRequestHandler implementation and relevant cleanup.

### CODE REVIEW

File: src/datacollector/EventHandlers/DataCollectionRequestHandler.cs [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
     using System.Threading;
     using System.Threading.Tasks;
 
     using Microsoft.VisualStudio.TestPlatform.Common.DataCollection;
+    using Microsoft.VisualStudio.TestPlatform.Common.DataCollection.Interfaces;
     using Microsoft.VisualStudio.TestPlatform.Common.DataCollector;
     using Microsoft.VisualStudio.TestPlatform.Common.DataCollector.Interfaces;
     using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
-    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.DataCollection.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.DataCollection;
```
Comment (AbhitejJohn) [inline src/datacollector/EventHandlers/DataCollectionRequestHandler.cs]: >DataCollection [](start = 69, length = 14)

Strange that there is a DataCollection in CommunicationUtilities. We plan to clean this up as well?

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestSettingsProviderPluginInformation.cs [hunk 2]
```diff
@@ -8,6 +8,8 @@ namespace Microsoft.VisualStudio.TestPlatform.Common.ExtensionFramework.Utilitie
     using System.Linq;
     using System.Reflection;
 
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel;
```
Comment (AbhitejJohn) [inline src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestSettingsProviderPluginInformation.cs]: >Microsoft.VisualStudio.TestPlatform.ObjectModel; [](start = 10, length = 48)

Why is this needed? Dont see anything added to OM.

File: src/Microsoft.TestPlatform.Common/Interfaces/Engine/ITestCaseEventsHandler.cs [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+
+    using Microsoft.VisualStudio.TestPlatform.Common.Interfaces.Engine;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+    using Microsoft.VisualStudio.TestPlatform.ObjectModel.DataCollection;
 
     /// <summary>
-    /// The Test Case level events.
+    /// The Test Case level events handler.
     /// </summary>
     public interface ITestCaseEventsHandler
```
Comment (AbhitejJohn) [inline src/Microsoft.TestPlatform.Common/Interfaces/Engine/ITestCaseEventsHandler.cs]: >ITestCaseEventsHandler [](start = 21, length = 22)

What exactly is the responsibility of this interface? Looks muddled up and is starving for documentation.

File: src/Microsoft.TestPlatform.Common/Interfaces/Engine/ITestRunCache.cs [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 
     using Microsoft.VisualStudio.TestPlatform.ObjectModel;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
-    using System;
 
     /// <summary>
     /// The cache for test execution information.
     /// </summary>
-    internal interface ITestRunCache : IDisposable
+    public interface ITestRunCache : IDisposable
```
Comment (AbhitejJohn) [inline src/Microsoft.TestPlatform.Common/Interfaces/Engine/ITestRunCache.cs]: >public [](start = 4, length = 6)

why?

File: src/Microsoft.TestPlatform.CommunicationUtilities/Microsoft.TestPlatform.CommunicationUtilities.csproj [hunk 5]
```diff
@@ -19,7 +19,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json">
-      <Version>8.0.3</Version>
+      <Version>9.0.1</Version>
```
Comment (AbhitejJohn) [inline src/Microsoft.TestPlatform.CommunicationUtilities/Microsoft.TestPlatform.CommunicationUtilities.csproj]: There was a reason this was made 8.0.3 IIRC. Tagging @mayankbansal018 for context.

**Comment by AbhitejJohn** [comment]
>     public sealed class TestRequestSender : ITestRequestSender

Does this class make sense in the engine? Common might be a better place?

---
Refers to: src/Microsoft.TestPlatform.CrossPlatEngine/TestRequestSender.cs:26 in 70b8d02. [](commit_id = 70b8d022ad114e0188064709819976bc983cbba6, deletion_comment = False)

File: src/Microsoft.TestPlatform.ObjectModel/Utilities/XmlRunSettingsUtilities.cs [hunk 6]
```diff
@@ -206,6 +206,23 @@ public static bool IsDataCollectionEnabled(string runSettingsXml)
         }
 
         /// <summary>
+        /// Returns whether data collection is enabled in the parameter settings xml or not
+        /// </summary>
+        /// <param name="runSettingsXml"> The run Settings Xml. </param>
+        /// <returns> True if data collection is enabled. </returns>
+        public static bool IsInProcDataCollectionEnabled(string runSettingsXml)
```
Comment (AbhitejJohn) [inline src/Microsoft.TestPlatform.ObjectModel/Utilities/XmlRunSettingsUtilities.cs]: >public [](start = 8, length = 6)

Do external customers need this? internal?

File: src/Microsoft.TestPlatform.CommunicationUtilities/Microsoft.TestPlatform.CommunicationUtilities.csproj (hunk 5 above)
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.CommunicationUtilities/Microsoft.TestPlatform.CommunicationUtilities.csproj]: I believe communication utilities is loaded inside VS IDE, which takes a dependency on 8.0.3, having 9.0.1 will break VS functionalities,
Adding Arun @codito for confirmation.

File: src/datacollector/EventHandlers/DataCollectionRequestHandler.cs (hunk 1 above)
Comment (harshjain2) [inline src/datacollector/EventHandlers/DataCollectionRequestHandler.cs]: Not all the DataCollection related cleanup is part of this PR. There will be a separate PR for datacollectors.

File: src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestSettingsProviderPluginInformation.cs (hunk 2 above)
Comment (harshjain2) [inline src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestSettingsProviderPluginInformation.cs]: `SettingsNameAttribute` is used in this class which is declared in ObjectModel.

**Comment by codito** [comment]
We decided to bring in responsibility of arranging the test case events in correct order (there is a mismatch on how xunit sends the events v/s rest of adapters) into TestExecutionRecorder. Rest of the changes we will bring in later.
